src/clients/SpokePoolClient.ts

```diff
@@ -725,11 +725,17 @@ export class SpokePoolClient extends BaseAbstractClient {
       for (const event of speedUpEvents) {
         const speedUp: SpeedUp = { ...spreadEvent(event.args), originChainId: this.chainId };
         assign(this.speedUps, [speedUp.depositor, speedUp.depositId], [speedUp]);
-      }
 
-      // Traverse all deposit events and update them with associated speedups, If they exist.
-      for (const deposits of Object.entries(this.depositHashes)) {
```

Comment:
I wonder if it would make sense to do this in two loops - i.e. first ensure all speedUp events get assigned to `this.speedUps`, and then have a 2nd loop over `speedUpEvents` that evaluates for each relevant deposit.

The reason I'm curious about this is that `this.appendMaxSpeedUpSignatureToDeposit()` expects that `this.speedUps` is complete. I'm not sure whether that creates an issue or not, but there is one test failing as a result of this change :thinking:

Reply:
nb. the above would still save a lot of loops, because we expect that nDeposits >> nSpeedUps.

Reply:
right, good point

Reply:
17dbca4dc128b1785fa4f97979029903319a51ed

Reply:
After offline discussions I've decided that its better to not support looping mode, which isn't used by anyone in prod IIUC, and maintain cleaner code